File: corporate-reporting/src/types/RiskMitigationAction.ts
import { IUser } from "./User";
import { IReportingEntityWithDeliveryDates, ReportingEntityWithDeliveryDates } from "./ReportingEntity";
import { DayOfWeek } from 'office-ui-fabric-react';
import { ReportingFrequency } from "../refData/ReportingFrequency";

export interface IRiskMitigationAction extends IReportingEntityWithDeliveryDates {
    Description: string;
    RiskID: number;
    RiskMitigationActionCode: number;
    OwnerUserID: number;
    ActionIsOngoing: boolean;
    OngoingActionReviewFrequency: ReportingFrequency;
    OngoingActionReviewDueDay: DayOfWeek;
    OngoingActionReviewStartDate: Date;
    NextReviewDate?: Date;
    OwnerUser?: IUser;
}

export class RiskMitigationAction extends ReportingEntityWithDeliveryDates implements IRiskMitigationAction {
    public Description = '';
    public RiskID = null;
    public RiskMitigationActionCode = null;
    public OwnerUserID = null;
    public ActionIsOngoing = false;
    public OngoingActionReviewFrequency = null;
    public OngoingActionReviewDueDay = null;
    public OngoingActionReviewStartDate = null;
    public Contributors = [];
}

File: ORB.Core/Models/RiskMitigationAction.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public partial class RiskMitigationAction : ReportingSubEntity
    {
        public RiskMitigationAction()
        {
        }

        public int RiskID { get; set; }
        public int? RiskMitigationActionCode { get; set; }
        public DateTime? BaselineDate { get; set; }
        public DateTime? ForecastDate { get; set; }
        public DateTime? ActualDate { get; set; }
        public int? OwnerUserID { get; set; }
        public bool? ActionIsOngoing { get; set; }
        public byte? OngoingActionReviewFrequency { get; set; }
        public byte? OngoingActionReviewDueDay { get; set; }
        public DateTime? OngoingActionReviewStartDate { get; set; }

        public virtual User OwnerUser { get; set; }
    }
}
